src/V3Timing.cpp

```diff
@@ -476,28 +518,96 @@ class TimingVisitor final : public VNVisitor {
-        if (m_classp) nodep->v3warn(E_UNSUPPORTED, "Unsupported: event controls in methods");
-        auto* const sensesp = m_finder.getSenTree(nodep->sensesp());
-        nodep->sensesp()->unlinkFrBack()->deleteTree();
-        // Get this sentree's trigger scheduler
+        // Do not allow waiting on local named events, as they get enqueued for clearing, but can
+        // go out of scope before that happens
+        if (nodep->sensesp()->exists<AstNodeVarRef>([](const AstNodeVarRef* refp) {
+                AstBasicDType* const dtypep = refp->dtypep()->basicp();
+                return dtypep && dtypep->isEvent() && refp->varp()->isFuncLocal();
+            })) {
+            nodep->v3warn(E_UNSUPPORTED, "Unsupported: waiting on local event variables");
+        }
         FileLine* const flp = nodep->fileline();
-        // Replace self with a 'co_await trigSched.trigger()'
-        auto* const triggerMethodp = new AstCMethodHard{
-            flp, new AstVarRef{flp, getCreateTriggerSchedulerp(sensesp), VAccess::WRITE},
-            "trigger"};
-        triggerMethodp->dtypeSetVoid();
-        // Add debug info
-        triggerMethodp->addPinsp(createEventDescription(sensesp));
-        triggerMethodp->addPinsp(new AstText{flp, '"' + flp->filename() + '"'});
-        triggerMethodp->addPinsp(new AstText{flp, cvtToStr(flp->lineno())});
-        // Create the co_await
-        auto* const awaitp = new AstCAwait{flp, triggerMethodp, sensesp};
-        awaitp->statement(true);
-        // Relink child statements after the co_await
-        if (nodep->stmtsp()) {
-            AstNode::addNext<AstNode, AstNode>(awaitp, nodep->stmtsp()->unlinkFrBackWithNext());
+        // Relink child statements after the event control
+        if (nodep->stmtsp()) nodep->addNextHere(nodep->stmtsp()->unlinkFrBackWithNext());
+        if (needDynamicTrigger(nodep->sensesp())) {
+            // Create the trigger variable and init it with 0
+            AstVarScope* const trigvscp
+                = createTemp(flp, m_dynTrigNames.get(nodep), nodep->findBitDType(), nodep);
+            auto* const initp = new AstAssign{flp, new AstVarRef{flp, trigvscp, VAccess::WRITE},
+                                              new AstConst{flp, AstConst::BitFalse{}}};
+            nodep->addHereThisAsNext(initp);
+            // Await the eval step with the dynamic trigger scheduler. First, create the method
+            // call
+            auto* const evalMethodp = new AstCMethodHard{
+                flp, new AstVarRef{flp, getCreateDynamicTriggerScheduler(), VAccess::WRITE},
+                "evaluation"};
+            evalMethodp->dtypeSetVoid();
+            // Add debug info
+            auto* const sensesp = nodep->sensesp();
+            evalMethodp->addPinsp(createEventDescription(sensesp));
+            evalMethodp->addPinsp(new AstText{flp, '"' + flp->filename() + '"'});
+            evalMethodp->addPinsp(new AstText{flp, cvtToStr(flp->lineno())});
```

Comment:
Does this cause a data escape when using --protect?  Similar elsewhere (maybe is similar bug in master too)

Reply:
Yes, if it's built with `VL_DEBUG`, it leaks the Verilog filename and line number. I'll open a PR that replaces these with sth like `<protected>:0` if `--protect-lib` is given.